src/main/java/mate/academy/springboot/datajpa/repository/ProductRepository.java

```diff
@@ -0,0 +1,14 @@
+package mate.academy.springboot.datajpa.repository;
+
+import java.math.BigDecimal;
+import java.util.List;
+import mate.academy.springboot.datajpa.model.Product;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface ProductRepository extends JpaRepository<Product, Long> {
+    List<Product> findAllByPriceBetween(BigDecimal firstPrice, BigDecimal secondPrice);
+
+    List<Product> getAllByCategories(List<Long> categories);
```

Comment:
I've got:
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'productController' defined in file [C:\java\projects\jv-springboot-data-jpa-delete\target\classes\mate\academy\springboot\datajpa\controller\ProductController.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'productServiceImpl' defined in file [C:\java\projects\jv-springboot-data-jpa-delete\target\classes\mate\academy\springboot\datajpa\service\impl\ProductServiceImpl.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'productRepository' defined in mate.academy.springboot.datajpa.repository.ProductRepository defined in @EnableJpaRepositories declared on JpaRepositoriesRegistrar.EnableJpaRepositoriesConfiguration: Invocation of init method failed; nested exception is org.springframework.data.repository.query.QueryCreationException: Could not create query for public abstract java.util.List mate.academy.springboot.datajpa.repository.ProductRepository.getAllByCategories(java.util.List)! Reason: Failed to create query for method public abstract java.util.List mate.academy.springboot.datajpa.repository.ProductRepository.getAllByCategories(java.util.List)! No property categories found for type Product!; nested exception is java.lang.IllegalArgumentException: Failed to create query for method public abstract java.util.List mate.academy.springboot.datajpa.repository.ProductRepository.getAllByCategories(java.util.List)! No property categories found for type Product!